### PR TITLE
Move onyx from community.network to be mellanox.onyx collection

### PR DESCRIPTION
**Pull request by samerd**
Signed-off-by: Samer Deeb <samerd@mellanox.com>

##### SUMMARY

Remove all references to onyx from collection towards merging to a new collection and redirect to the new collection.

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by felixfontein** [comment]
And finally, this PR needs a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to) with `removed_features`.

File: meta/runtime.yml [hunk 1]
```diff
@@ -318,3 +318,19 @@ plugin_routing:
         removal_version: 2.0.0
         warning_text: The onyx_wjh module has been moved to the mellanox.onyx collection
       redirect: mellanox.onyx.onyx_wjh
+  cliconf:
+    onyx:
+      redirect: mellanox.onyx.onyx
```
Comment (felixfontein) [inline meta/runtime.yml:323]:
```suggestion
      deprecation:
        removal_version: 2.0.0
        warning_text: The onyx cliconf plugin has been moved to the mellanox.onyx collection
      redirect: mellanox.onyx.onyx
```
Same for the others.

File: changelogs/fragments/83-remove-onyx.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+removed_features:
+- "onyx - remove support for all onyx plugins because of migration to mellanox.onyx collection."
```
Comment (felixfontein) [inline changelogs/fragments/83-remove-onyx.yml:2]:
```suggestion
- "onyx - all onyx modules and plugins have been moved to the mellanox.onyx collection. Redirects have been added that will be removed in community.network 2.0.0 (https://github.com/ansible-collections/community.network/pull/83)."
```

File: meta/runtime.yml (hunk 1 above)
Comment (samerd) [reply meta/runtime.yml:326]: fixed

File: changelogs/fragments/83-remove-onyx.yml (hunk 2 above)
Comment (samerd) [reply changelogs/fragments/83-remove-onyx.yml]: fixed

**Comment by gundalow** [comment]
@samerd thanks for this PR.
When do you think you'll have the new collection build, even if it's just a `0.1.0` release?

We are wondering if we want to delete this content only after the new collection is live.

**Comment by samerd** [comment]
@gundalow : I am working on it now.
I can create a 0.1.0 release today or tomorrow

**Comment by samerd** [comment]
@gundalow mellanox.onyx -0.1.0 collection has been published

**Comment by gundalow** [comment]
> @gundalow mellanox.onyx -0.1.0 collection has been published

@samerd Thank you, https://github.com/ansible-community/ansible-build-data/pull/11 means it will be included in the next `ansible` package

**Pull request merged**